### ISSUE TITLE
Submission Title Format not parsing any variable tokens for grouped fields

### ISSUE THREAD

**Issue by johndwells** [verbb/formie#128]
**Description**

I have configured a form's "Submission TItle Format" to use a number of fields in the form: First Name, Last Name, Email Address, and then Timestamp. However only the Timestamp is ever being parsed, all other tokens come up empty. All fields are required, and contain submitted information.

![Screen Shot 2020-09-21 at 11 21 18 AM](https://user-images.githubusercontent.com/134110/93793577-99452f80-fbfc-11ea-9690-f43d95cab3c5.png)

![Screen Shot 2020-09-21 at 11 21 46 AM](https://user-images.githubusercontent.com/134110/93793613-a6fab500-fbfc-11ea-9f45-889a643fa7dc.png)

**Additional info**
- Multi-page form: No
- Submission Method: Page Reload
- Client-side Validation: Yes
- Custom Form Templates: No

**Additional info**
- Plugin version: 1.2.15
- Craft version: 3.5.11.1
- Multi-site: Yes or No

**Comment by engram-design** [comment]
I can't seem to immediately replicate this: https://d.pr/i/5l3J5L and https://d.pr/i/GrleLR

However, I notice a tiny detail where it's including a colon `:` after the field name in the variable pill. Can you confirm what fields these are?

**Comment by engram-design** [comment]
As a debugging step, you can also use plain Twig shorthand in this field. Might be worth trying something like `{firstName} - {lastName} - {email}` (replacing those values with the field handles)

**Comment by johndwells** [comment]
I thought the colon might have been relevant, but no dice. And using the twig shorthand hasn't helped.

I decided to create a sample form using the Contact Form stencil, and that worked as expected, the submission title properly parsed the field values. Then I noticed that the contact form doesn't put any of the fields into "groups"; but I have my form split up into groups. So as a test I created another field on my original form, placed outside of a group, and that was parsed fine in the submission title.

Do I need some alternative syntax to access a field that's in a group?

**Comment by engram-design** [comment]
Ah, I thought groups might've been what you were using. Can you shoot through a screenshot of your field setup just so I can replicate? I did some tests with groups which also worked...

**Comment by johndwells** [comment]
Sure thing.... let me know if this is enough:

The Group of fields in the form designer:
![Screen Shot 2020-09-22 at 5 32 08 AM](https://user-images.githubusercontent.com/134110/93872126-4d929480-fc95-11ea-8bdc-0d5018484caa.png)

Advanced tab of the group field:
![Screen Shot 2020-09-22 at 5 32 27 AM](https://user-images.githubusercontent.com/134110/93872125-4d929480-fc95-11ea-8dca-7f3cef9bfcbe.png)

Advanced tab of the email field:
![Screen Shot 2020-09-22 at 5 32 36 AM](https://user-images.githubusercontent.com/134110/93872124-4cf9fe00-fc95-11ea-868b-7738ab403e77.png)

Manually entering the twig syntax:
![Screen Shot 2020-09-22 at 5 32 50 AM](https://user-images.githubusercontent.com/134110/93872123-4c616780-fc95-11ea-9051-f8fbc3486cc6.png)

Refreshing the page showing that `{email}` gets properly turned into a token:
![Screen Shot 2020-09-22 at 5 33 09 AM](https://user-images.githubusercontent.com/134110/93872122-4c616780-fc95-11ea-8070-50d677cccd0c.png)

Available fields in the token dropdown:
![Screen Shot 2020-09-22 at 5 34 16 AM](https://user-images.githubusercontent.com/134110/93872116-4bc8d100-fc95-11ea-9fcb-c04e4535e099.png)

Result of a front end submission, no parse:
![Screen Shot 2020-09-22 at 5 33 36 AM](https://user-images.githubusercontent.com/134110/93872120-4c616780-fc95-11ea-9e03-79362d076479.png)

Thanks Josh!

**Comment by engram-design** [comment]
Perfect, I can replicate this now. Looks like this might've been an oversight on my part not catering for group fields.

**Comment by johndwells** [comment]
Ah, ok great, relieved to have gotten to the bottom of it. Thanks Josh!

**Comment by engram-design** [comment]
Just fixed for the next release.

To get the fix early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-3 as 1.2.16",
  "...": "..."
}
```

Then run `composer update`.

**Comment by johndwells** [comment]
Yes! That works a charm. Thanks Josh.

I'm curious, what's the twig shorthand equivalent of the token?

**Comment by engram-design** [comment]
So for a Group or Repeater field, it actually returns an element query, because they use a NestedField element - a little bit similar to a Matrix field. So the the variable pill is actually using `groupField.one().innerField ?? null` - you could use this in shorthand twig.

To illustrate: https://d.pr/i/jBsApL

**Comment by engram-design** [comment]
Fixed in [1.2.17](https://github.com/verbb/formie/releases/tag/1.2.17)
